src/main/java/ro/siit/trips/controllers/TripController.java

```diff
@@ -0,0 +1,190 @@
+    private TripService tripService;
+
+
+    @RequestMapping(value = "/trip")
+    public String journey(Model model) {
+        Trip trip = new Trip();
+        model.addAttribute("trip", trip);
+        return "addtrip";
+    }
+
+    @RequestMapping(value = "/addtrip", method = RequestMethod.POST)
+    public ModelAndView addTrip(@RequestParam("file") MultipartFile file, @RequestParam("file2") MultipartFile file2, @Valid @ModelAttribute Trip trip, BindingResult bindingResult) {
+        ModelAndView model = new ModelAndView("successtrip");
+        ModelAndView model2 = new ModelAndView("/addtrip");
+
+        if (bindingResult.hasErrors()) {
+            return model2;
+        }
+        Person loggedPerson = personRepository.findByUserName(SecurityContextHolder.getContext().getAuthentication().getName());
+
+
+        trip.setPerson(loggedPerson);
+        Trip savedTrip = tripRepository.save(trip);
+
+
+        String uuid1 = UUID.randomUUID().toString();
+        savedTrip.setPhoto1(uuid1);
+        fileService.store(savedTrip.getPhoto1(), file);
+        model.addObject("uuid1", savedTrip.getPhoto1());
+
+        String uuid2 = UUID.randomUUID().toString();
+        savedTrip.setPhoto2(uuid2);
+        fileService.store(savedTrip.getPhoto2(), file2);
+        model.addObject("uuid2", savedTrip.getPhoto2());
+        tripRepository.save(savedTrip);
+        return model;
+    }
+
+    @RequestMapping(path = "/trips")
+    public ModelAndView getTrips(@RequestParam(name = "id", required = false) Long id) {
+        ModelAndView model = new ModelAndView("tripTemplate");
+        ModelAndView model2 = new ModelAndView("redirect:/trip");
+        Person loggedPerson = personRepository.findByUserName(SecurityContextHolder.getContext().getAuthentication().getName());
+        if (loggedPerson.getTrips().size() == 0) {
+            return model2;
+        }
+        Trip currentTrip = null;
+        if (id != null) {
+            for (Trip trip : loggedPerson.getTrips()) {
+                if (trip.getId().equals(id)) {
+                    currentTrip = trip;
+                    System.out.println("Found:" + trip);
+                    break;
+                }
+            }
+        }
+        if (currentTrip == null) {
+            currentTrip = loggedPerson.getTrips().iterator().next();
+
+        }
+        model.addObject("trips", loggedPerson.getTrips());
+        model.addObject("selectedId", currentTrip.getId());
+        model.addObject("currentTrip", currentTrip);
+        return model;
+
+    }
+
+    @RequestMapping(value = "/delete", method = RequestMethod.POST)
+    public ModelAndView delete(@ModelAttribute Trip trip, @RequestParam(name = "id", required = false) Long id) {
+        ModelAndView model = new ModelAndView("tripTemplate");
+        tripRepository.deleteById(trip.getId());
+        Person loggedPerson = personRepository.findByUserName(SecurityContextHolder.getContext().getAuthentication().getName());

```

Comment:
don't do this here... just delete the trip here, then do a redirect to the /trips page so that you don't duplicate code... less chances of bugs&errors